src/Ema/Route/Generic/Verification.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE UndecidableInstances #-}
+
+-- TODO: Consider moving to Internal.Verification instead?
+module Ema.Route.Generic.Verification (
+  type VerifyModels,
+  type VerifyRoutes,
+) where
+
+import Data.Type.Bool (If, type (||), type (&&))
+import Data.Type.Equality (type (==))
+import GHC.Generics qualified as GHC
+import GHC.TypeLits (Symbol, type (-))
+import Type.Errors.Pretty (TypeError)
+import Type.Errors.Pretty qualified as P
+
+-- Index into the nth field of a single-constructor generic type rep, returning its type
+type family Indexed (i :: Nat) (xs :: Type) :: Type where
+  Indexed n (GHC.D1 _ (GHC.C1 _ fields) _) = 
+    Indexed n (fields ())
+  Indexed 1 ((GHC.S1 _ (GHC.K1 _ t) GHC.:*: nxt) _) = 
+    t
+  Indexed n ((GHC.S1 _ _ GHC.:*: nxt) _) = 
+    Indexed (n - 1) (nxt ())
+  Indexed 1 (GHC.S1 _ (GHC.K1 _ t) _) =
+    t
+  Indexed 0 (GHC.S1 _ _ _) =
+    TypeError ("Type rep indexing: generic selector indexing starts at 1" P.% "")
+  Indexed n (GHC.S1 _ _ _) =
+    TypeError ("Type rep indexing: out of bounds index " P.<> n)
+  Indexed _ _ = 
+    TypeError ("Type rep indexing: multiple constructors" P.% "") 
+
+-- | Extract the type of a field of name @s@ from a generic type representation @t@
```

Comment:
An example would be useful here as well.

Reply:
Added examples as of [a873e37](https://github.com/EmaApps/ema/pull/120/commits/a873e371390da804fbdbb14d6074b1538b3e4031)